oneDNN SYCL support for LRN primitive

saisrujana opened the pull request:
# Description

Introducing SYCL backend for the oneDNN LRN (Local Response Normalization) primitive. This contains support for LRN functionality in both forward and backward propagations.

# Supported scope
Supported Data Types:
Forward/ Backward : f32, bf16, f16

# Tested Hardware
Nvidia Tesla T4

## General

- [x] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [x] Have you formatted the code using clang-format?

## Testing Scope
`benchdnn : passed`

## Test Output Files
[test_lrn_gpu.txt](https://github.com/oneapi-src/oneDNN/files/10941928/test_lrn_gpu.txt)
[test_lrn_ci.txt](https://github.com/oneapi-src/oneDNN/files/10941931/test_lrn_ci.txt)
[test_lrn_all.txt](https://github.com/oneapi-src/oneDNN/files/10941932/test_lrn_all.txt)

densamoilov commented on src/gpu/nvidia/sycl_cuda_engine.cpp:
```diff
@@ -223,6 +224,14 @@ constexpr dnnl::impl::impl_list_item_t sycl_cuda_impl_list[] = {
         INSTANCE(cudnn_lrn_fwd_t)
         INSTANCE(cudnn_lrn_bwd_t)
 
+        INSTANCE(sycl::ref_sycl_lrn_fwd_t<data_type::f32>)
```
Please refactor the code to avoid instantiating the primitive for each data type. You can use eltwise primitive for reference.

saisrujana replied:
Done. Please check https://github.com/oneapi-src/oneDNN/pull/1605

densamoilov reviewed (changes requested):
Please address the comments.

saisrujana commented:
> Please address the comments.

Due to the deletion of head repository, Commits weren't reflected in this PR.  Hence raised a new PR addressing your comments. Please check https://github.com/oneapi-src/oneDNN/pull/1605 . Thanks!
